File: tests/fixtures/6588/test/parse.py
class Parse:
    def parse_and_solve(self, sol, input: str) -> str:
        lines = input.strip().split('\n')
        results = []
        for line in lines:
            n = int(line.strip())
            if n == 0:
                break
            results.append(sol.solve(n))
        return '\n'.join(results)

File: tests/fixtures/6588/test/Parse.java
public class Parse implements ParseAndCallSolve {
    @Override
    public String parseAndCallSolve(Solution sol, String input) {
        StringBuilder sb = new StringBuilder();
        String[] lines = input.trim().split("\n");
        boolean first = true;
        for (String line : lines) {
            int n = Integer.parseInt(line.trim());
            if (n == 0) break;
            if (!first) sb.append("\n");
            sb.append(sol.solve(n));
            first = false;
        }
        return sb.toString();
    }
}
